src/security/dependency-track/steps/generate/generate-dotnet-sboms.step.yaml

```
@@ -0,0 +1,121 @@
+        New-Item -ItemType Directory -Path "$(Agent.TempDirectory)/sbom/dotnet" -Force | Out-Null
+        
+        # Set error handling to stop on errors
+        $ErrorActionPreference = 'Stop'
+        
+        # Retrieve the JSON string containing .NET project paths
+        $DotnetProjectsJson = '$(resolveDotnetProjects.dotnetProjectsJson)'
+        $OutputDir = '$(Agent.TempDirectory)/sbom/dotnet'
+
+        # Initialize an array to store project paths
+        $projPaths = @()
+        
+        # Attempt to parse the JSON string into an array of project paths
+        try { 
+          $projPaths = $DotnetProjectsJson | ConvertFrom-Json 
+        } catch { 
+          Write-Host "##vso[task.logissue type=warning]Invalid dotnetProjectsJson: $DotnetProjectsJson"; 
+          $projPaths = @() 
+        }
+        
+        # Exit if no project paths are provided
+        if ($projPaths.Count -eq 0) { 
+          Write-Host "##vso[task.logissue type=warning].NET project list empty; skipping"; 
+          exit 0 
+        }
+        
+        # Define output format and file extension for SBOMs
+        $format = 'Json'; 
+        $ext = 'json'; 
+        $ok = 0; 
+        $fail = 0
+        
+        # Iterate over each project path to generate SBOMs
+        foreach ($p in $projPaths) {
+          $base = [System.IO.Path]::GetFileNameWithoutExtension($p)
+          $fileName = "{0}-sbom.{1}" -f $base, $ext
+        
+          # Log the current project being processed
+          Write-Host "##[group].NET → $p"
+          try {
+            # Run CycloneDX to generate the SBOM
+            dotnet CycloneDX "$p" -o "$OutputDir" -F "$format" -fn "$fileName"
+        
+            # Check if the SBOM file was successfully created
+            if ($LASTEXITCODE -ne 0) { 
+              throw "CycloneDX exit code: $LASTEXITCODE" 
+            }
+            if (Test-Path (Join-Path $OutputDir $fileName)) { 
```

What's the purpose behind this logic (and tracking the successes and failures in general)? If CycloneDX fails (i.e. non-zero exit code) it will throw an error. Under what circumstances will CycloneDX return a zero exit code but not produce an SBOM?

---

The tracking is done for logging purposes again. It just improves observability and allows us to fail fast rather than trying to carry on with the pipeline with an empty set of SBOMs.
Don't get me wrong - it should be rare for CycloneDX to return a zero exit code but not produce an SBOM but it is ultimately a dependency in itself and outside of our control so isn't it better to be safe?